src/pysource_minimize/_minimize.py

```diff
@@ -105,6 +110,87 @@ def source_checker(new_ast):
         source_checker,
         progress_callback=progress_callback,
         retries=retries,
+        strategies=strategies,
     )
 
     return unparse(minimized_ast)
+
+
+def minimize(
+    source: str,
+    checker: Callable[[str], bool],
+    *,
+    progress_callback: Callable[[int, int], object] = lambda current, total: None,
+    retries: int = 1,
+    compilable=True,
+) -> str:
+    """
+    minimizes the source code
+
+    Args:
+        source: the source code to minimize
+        checker: a function which gets the source and returns `True` when the criteria is fulfilled.
+        progress_callback: (deprecated) function which is called everytime the source gets a bit smaller.
+        retries: the number of retries which should be performed when the ast could be minimized (useful for non deterministic issues)
+        compilable: make sure that the minimized code can also be compiled and not just parsed.
+
+    Warning:
+        `progress_callback` is deprecated and should be implemented inside in `checker` where you can use the `len(source_code)`
+        to calculate the progress.
+
+    returns the minimized source
+    """
+    return _minimize_source(
+        source,
+        checker,
+        progress_callback=progress_callback,
+        retries=retries,
+        compilable=compilable,
+    )
+
+
+def minimize_all(
+    sources: dict[Path, str],
+    checker: Callable[[dict[Path, str | None], Path], bool],
+    *,
+    retries: int = 1,
+    compilable=True,
+) -> dict[Path, str | None]:
+    """
+    minimizes multiple source codes.
+
+    Args:
+        sources: the source code to minimize
+        checker: a function which gets the source and returns `True` when the criteria is fulfilled.
+        retries: the number of retries which should be performed when the ast could be minimized (useful for non deterministic issues)
+        compilable: make sure that the minimized code can also be compiled and not just parsed.
+
+    Returns:
+        a dict with the minimized sources. The values are `None` when the source file should be deleted
+    """
```

Comment:
@pawamoy, what do you think about this interface? It works almost like minimize() with the exception that you give it a dict of source files with the pathes as keys. The sources can become None which means that this source file is deleted.

You can take a loot at test_minimize_all for example usage.